ergotree-interpreter/src/eval/subst_const.rs

```diff
@@ -56,30 +35,57 @@ impl Evaluable for SubstConstants {
                 }
                 (elem_tpe, items_const)
             } else {
-                return Err(EvalError::Misc(String::from(
+                return Err(EvalError::Misc(
                     "SubstConstants: expected evaluation of `new_values` be of type `Coll[_]`, got \
-                    _ instead",
-                )));
+                    _ instead".into()
+                ));
             };
 
+        if new_constants.len() != positions.len() {
+            return Err(EvalError::Misc(
+                "SubstConstants: lengths of `positions` and `new_values` differ".into(),
```

Comment:
Let's show the exact lengths in the error message.

Reply:
Done.